### DISCUSSION TITLE
How to enable ftps for private ip ?

### DISCUSSION THREAD

**Discussion by almgwary** [QuorumDMS/ftp-srv#311] [Q&A]
Kindly provide example how to enable ftps for private server running on private ip, I've tried to use it but it is not connecting or showing any error messages.

```



   const tlsOptions = {
     key: fs.readFileSync('./ssl/private.key'),
     cert: fs.readFileSync('./ssl/cert.pem'),
     ca:fs.readFileSync('./ssl/cert.pem'),
 }


const ftpServer = new FtpSrv({
    url: `ftp://${hostname}:${ftpPort}`,
    pasv_url: `ftp://0.0.0.0`,
    pasv_url: resolverFunction,
    tls: false,
    pasv_min:21000,
    pasv_max:21010,
    anonymous: true,
    greeting: ["Hello", "hi"]
}, './ftp');



```

**Comment by matt-forster** [comment]
Hey @almgwary, I’m away from my desktop right now so of limited help - but I am converting this to a discussion because I think that is more appropriate. I do have a couple of suggestions looking at your code.

I see you’ve created the TLS options object - but you are not passing it into the constructor. You should pass it into the `tls` attribute. 
You’ll also want to use FTPS in the URL.

The PASV url should work locally, but beware of using 0.0.0.0 in any other environment than your desktop. You’ll want that to be the actual machine IP on the network if you are connecting from a different device - or the exposed IP if elsewhere.

When I get back next week I’ll try to get a proper example working, and then we can add it to the readme. TLS hasn’t had as much attention as it is needed unfortunately.

**Comment by almgwary** [comment]
Yes already tried to use `tls:tlsOptions` but it is not working.
I'll wait for you working example when you have time, Thank you
